lib/charms/grafana_k8s/v0/grafana_metadata.py

```diff
@@ -0,0 +1,250 @@
+"""grafana_metadata.
+
+This implements provider and requirer sides of the grafana-metadata interface, which is used to communicate information
+about an Grafana installation such as its url and UID.
+
+## Usage
+
+### Requirer
+
+To add this relation to your charm as a requirer, add the following to your `charmcraft.yaml` or `metadata.yaml`:
+
+```yaml
+requires:
+  grafana-metadata:
+    # The example below uses the API for when limit=1.  If you need to support multiple related applications, remove
+    # this and use the list-based data accessor method.
+    limit: 1
+    interface: grafana_metadata
+```
+
+To handle the relation events in your charm, use `GrafanaMetadataRequirer`.  That object handles all relation events for
+this relation, and emits a `DataChangedEvent` when data changes the charm might want to react to occur.  To set it up,
```

Comment:
```suggestion
this relation, and emits a `DataChangedEvent` that the charm can observe.  To set it up,
```

as a non-native speaker, this sentence jumbled me up quite seriously

Reply:
as a native speaker, this sentence jumbled me up as well :D

Reply:
Reworded and I think it should be clearer, but please reopen if I'm wrong.